src/main/java/fr/moribus/imageonmap/commands/maptool/ExploreCommand.java

```diff
@@ -55,31 +56,35 @@
 public class ExploreCommand extends IoMCommand {
     @Override
     protected void run() throws CommandException {
-        if (args.length < 1) {
-            Gui.open(playerSender(), new MapListGui(playerSender()));
-        } else {
-            if (Permissions.LISTOTHER.grantedTo(sender)) {
-                String name = args[0];
-                Player sender = playerSender();
-                offlinePlayerParameter(0, uuid -> {
-                    if (uuid == null) {
-                        try {
-                            throwInvalidArgument(I.t("Player not found."));
-                        } catch (CommandException e) {
-                            PluginLogger.error("CommandException " + e);
-                            return;
-                        }
-                    }
+        ArrayList<String> arguments = getArgs();
+        if (arguments.size() > 1) {
+            warning(I.t("Too many parameters! Usage: /maptool explore [playername]"));
+            return;
+        }
+        final String playerName;
 
-                    OfflinePlayer offlinePlayer = Bukkit.getOfflinePlayer(uuid);
-                    if (offlinePlayer != null) {
-                        Gui.open(sender, new MapListGui(offlinePlayer, name));
-                    } else {
-                        PluginLogger.warning(I.t("Can't find player"));
-                    }
-                });
+        final Player sender = playerSender();
+        if (arguments.size() == 1) {
+            if (!Permissions.LISTOTHER.grantedTo(sender)) {
+                info(sender, I.t("You can't use this command"));
+                return;
             }
+            playerName = arguments.get(0);
+        } else {
+            playerName = sender.getName();
         }
+
+        //TODO passer en static
+        ImageOnMap.getPlugin().getCommandWorker().offlineNameFetch(playerName, uuid -> {
+            if (uuid == null) {
+                info(sender, I.t("The player {0} does not exist.", playerName));
+                return;
+            }
+            OfflinePlayer offlinePlayer = Bukkit.getOfflinePlayer(uuid);
+            Gui.open(sender, new MapListGui(offlinePlayer,playerName));
```

Comment:
Idem, ajouter un `if (sender.isOnline())` ne mangerait pas de pain.

```suggestion
            if (sender.isOnline()) {
                Gui.open(sender, new MapListGui(offlinePlayer,playerName));
            }
```